### DISCUSSION TITLE
旧字体・異体字表記の単語の登録要望について

### DISCUSSION THREAD

**Discussion by tats-u** [google/mozc#1052] [Q&A] [answered]
一部の方から旧字体・異体字表記の単語登録要望がちらほらありますが、無条件に承認しますか？そうでない場合、どのようなルールで承認しますか？
きりがないと思いますし、「神︀」などコードポイントが複数ある漢字もあります。
また、大抵のユーザは利用しないので、単語によっては邪魔になりえます。Mozcのビルド後サイズも増大します。

「豫園」（登録済み）など一部例外はあります
また単漢字はOKです
場合によっては旧字体モード（旧字体単語も出す or しか出さない）などを作って対処しないといけないかもしれませんが、設定したいユーザは少数でしょう。また、コードポイント複数漢字はどれを出すべきかを考えないといけません。

神の旧字体：

- 神 (U+FA19) CJK互換漢字
- 神︀ (U+795E U+FE00) 新字体 + SVS
- 神󠄀 (U+795E U+E0100) 新字体 + IVS (Adobe Japan-1)
- 神󠄃 (U+795E U+E0103) 新字体 + IVS (MJ)

一番上は一番馴染みが深いですが、正規化で新字体にされる場合があります。

https://moji-memo.hatenablog.jp/entry/20130116/1358319668

下3つは一部アプリケーションでの取り扱いに難があり（特に一番下）、フォント次第では新字体で表示されます。
特に一番下はIPAmj明朝専用です。
Mozc・Google日本語入力では一番上しか入力できません。

**Comment by hiroyuki-komatsu** [comment, marked as answer]
First of all, we appreciate the reports from our contributors, which provide valuable insights from various perspectives.

However, as you mentioned, we need to prioritize requests to strike a balance between usability and coverage. Therefore, it's possible that some requests may not be addressed until we find that optimal balance. The implementation effort required and the potential impact on other conversion cases will also be factors in our decision-making.

Thank you for reminding us about the issues surrounding variation characters. We are aware of them, particularly the IVS issue you highlighted.

**Reply by tats-u** [reply]
There seems not to be no concrete criteria that you can state today yet.
Thank you for your answer.

**Comment by faketanaka** [comment]
個人的には、旧字体の単語が変換できればとても便利です。しかし、これを必要とする人は少数であるというのはおそらくは事実でしょう。
ATOKの「文語モード」のように、旧字体を変換候補に出す/出さないを切り替えられれば便利だと思います。
デフォルトでは旧字体は単漢字変換のみに出すようにし、「旧字体モード」をオンにした場合に限って単漢字変換以外でも候補に出す、というのも手でしょう。

しかしながら、これを実装するとなると大掛かりな作業が必要でしょう。実装していただけたらとてもありがたいですが、優先順位が下がるのは仕方ないことですね。

**Comment by faketanaka** [comment]
先日、Gboardを使う機会がありました。Gboardでは、旧字体の単語（単漢字ではなく）がかなりたくさん出てくる印象を受けます。MozcとGboardでは辞書データが異なるのは重々承知しておりますが、目指す姿や開発方針も異なるという認識でいいのでしょうか。
